client/components/domains/registrant-extra-info/fr-validate-contact-details.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { get, reduce, set } from 'lodash';
+import validatorFactory from 'is-my-json-valid';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import validationSchema from './fr-schema';
+
+const validate = validatorFactory( validationSchema, { greedy: true } );
+const debug = debugFactory( 'calypso:components:domains:registrant-extra-info:validation' );
+
+/*
+ * @returns errors by field, like: { 'extra.field: name, errors: [ string ] }
+ */
+export default function validateContactDetails( contactDetails ) {
+	// Populate validate.errors
+	validate( contactDetails );
+	validate.errors && debug( validate.errors );
+
+	return reduce(
+		validate.errors,
+		( acc, { field, message } ) => {
+			// Drop 'data.' prefix
+			const path = String( field ).split( '.' ).slice( 1 );
```

Comment:
Maybe better to call it `key` or something? Btw, why the explicit conversion?

Reply:
The `String()`? It's to avoid an error if `field` is `undefined` or something surprising. That only happens if I screw up elsewhere, so it's not _strictly_ necessary, but errors wreck Calypso so hard that I try really hard not to leave openings.

I think `key` would be the right name if we were using it as a single string, like `{ 'extra.sirenSiret': 'validation failed' }` but `path` seems better for this array-of-keys format.  It's what Lodash uses, for example - https://lodash.com/docs/4.17.4#set `_.set( object, path, value )`.

I'm still not totally whether I'm doing this right. I was using it as a key for a while, but it's annoying to consume.